internal/protocol/message.go

```diff
@@ -229,12 +217,46 @@ func (m *Message) putNamedValues(values NamedValues) {
 			panic("unsupported value type")
 		}
 	}
+}
+
+// Encode the given driver values as binding parameters.
+func (m *Message) putNamedValues(values NamedValues) {
+	l := len(values)
+	if l > 255 {
+		// safeguard, should have been checked beforehand.
+		panic("too many parameters")
```

Comment:
Admittedly this isn't a convention elsewhere (so likely not an issue for this PR) but would it be better to have `"internal error: ..."` to highlight that the user isn't at-fault for these panics?